### ISSUE TITLE
Targeting self with a hostile weapon proc running interrupts auto attack

### ISSUE THREAD

**Issue by TheTinkeringIdiot** [project-rubika/issues#627]
**Describe the bug**
When fighting a mob while running a weapon proc nano, targeting self will lock auto-attack within a few cycles, after which the character cannot continue combat without canceling auto-attack and restarting. 

I tested this using both Keeper proc nano lines (Evangelical Reaper and Divine Sanctifier). I suspect it is the same for the Shade proc nanos, but those are broken per #617 and do not produce the proc effects. 

This may be related to #593, and it may be that the direct-damage component of the proc is causing an error when applied to self and halting auto-attack, but I am documenting here in cast it's something different. In earlier testing weapon procs (Edge of Tarasque) have no qualms about applying to self, and the spell effects for these Keeper nanos both call out target=target, so this seems slightly different from the other issue, even if related. 

**To Reproduce**
1. Be Keeper
2. Run Evangelical Reaper or Divine Sancitifier
3. Fight a target
4. Target self
5. Auto-attack halts but combat does not end

**Expected behavior**
Targeting self should not stop auto-attacking the fighting target

**Screenshots**
![keeper_self_target](https://github.com/project-rubika/issues/assets/96996793/71a211e1-749b-488a-8159-41efeb47d6f1)

**Character info:**

```
Tinkerkeep [220 Keeper]
------------------------------------
    assaultrifle 	2437
    submachinegun 	2449
    shotgun 	2441
    flingshot 	2415
    burst 	2448
    fullauto 	2480
    lr_multipleweapon 	2391
    distanceweaponinitiative 	118
    electricalengineering 	2398
    fieldquantumphysics 	2360
    pharmaceuticals 	2368
    nanoprogramming 	2356
    chemistry 	2418
    disarmtrap 	2372
    maxnanoenergy 	6736
    selectedtargettype 	0
    charstate 	0
    isfightingme 	1
    bodydevelopment 	2769
    runspeed 	2869
    nanoac 	4128
    twohandedgedweapons 	4289
    closecombatinitiative 	3767
    senseimprovement 	3610
    firstaid 	3758
    treatment 	3614
    materialmetamorphose 	2698
    biologicalmetamorphose 	3598
    psychologicalmodification 	3586
    materiallocation 	3777
    nanoenergypool 	2114
    adventuring 	2889
    brawl 	3779
    riposte 	3100
    sneakattack 	3076
    nanoprowessinitiative 	2446
    duck 	3682
    dodge 	3682
    evade 	3942
    computerliteracy 	3585
    parry 	2991
    fastattack 	3093
    onehandbluntweapons 	2793
    onehandedgedweapon 	2773
    piercing 	2767
    twohandbluntweapons 	2799
    meleeenergyweapon 	2737
    dimach 	2682
    meleemultiple 	2760
    petreq1 	14
    nanofocuslevel 	96
    materialcreation 	2682
    aimedshot 	2507
    mapnavigation 	2649
    perception 	2650
    psychology 	2649
    throwingknife 	2707
    tutoring 	2589
    driveair 	2661
    driveground 	2661
    drivewater 	2661
    weaponsmithing 	2707
    bowspecialattack 	2690
    mechanicalengineering 	2530
    concealment 	2466
    breakingentry 	2673
    swim 	2756
    physicalprowessinitiative 	2657
    bow 	2638
    rifle 	2452
    lr_energyweapon 	2420
    throwngrapplingweapons 	2643
    currentncu 	326
    specialization 	15
    side 	1
    martialarts 	2742
    grenade 	2383
    pistol 	2394
    strength 	1922
    stamina 	1922
    agility 	1663
    sense 	1455
    psychic 	1455
    intelligence 	1425
    flags 	8917569
    monsterscale 	100
    currentnano 	6736
    headmesh 	40111
    waitstate 	1
    cash 	1000
    gmlevel 	255
    expansion 	3
    breed 	4
    visualbreed 	4
    profession 	14
    visualprofession 	14
    level 	220
    sex 	1
    visualsex 	1
    health 	217005
    titlelevel 	7
    accountflags 	0
    equippedweapontype 	130
    EquippedRHWeapon 	130
    selectedtarget 	1283
    nextxp 	3067008
    xp 	2555840
    lastsaved 	2555840
    lastxp 	2555840
    maxncu 	2517
    playfieldtype 	0
    fatness 	1
    race 	1
    life 	117005
    monsterdata 	0
    VisualFlags 	31
    currentmovementmode 	3
    clan 	0
    healinterval 	1
    nanointerval 	1
    meleedamagemodifier 	433
    energydamagemodifier 	406
    projectiledamagemodifier 	394
    firedamagemodifier 	433
    colddamagemodifier 	394
    chemicaldamagemodifier 	426
    poisondamagemodifier 	426
    radiationdamagemodifier 	406
    amsmodifier 	250
    attackshield 	1
    projectileac 	23206
    energyac 	22786
    meleeac 	23206
    coldac 	23576
    fireac 	23800
    chemicalac 	23568
    radiationac 	23628
    poisonac 	23623
    shieldprojectileac 	103
    shieldmeleeac 	116
    shieldenergyac 	103
    shieldchemicalac 	116
    shieldradiationac 	103
    shieldcoldac 	116
    shieldnanoac 	80
    shieldfireac 	103
    shieldpoisonac 	116
    reflectmeleeac 	203
    reflectpoisonac 	203
    reflectfireac 	203
    reflectnanoac 	3
    reflectcoldac 	203
    reflectprojectileac 	203
    reflectenergyac 	203
    reflectchemicalac 	203
    reflectradiationac 	203
    nanodamagemodifier 	187
    team 	0
    dmsmodifier 	545
    xpmodifier 	10
    criticalincrease 	3
    nanodelta 	100028
    healmultiplier 	4
    nanodamagemultiplier 	4
    healdelta 	100143
    procnano4 	2
    procchance4 	267782
    rangeincreaserweapon 	37
    criticaldecrease 	3
    npcostmodifier 	-2
    aggdef 	100
    state 	0
    unarmedtemplateinstance 	0
    invaderskilled 	0
    killedbyinvaders 	0
    vp 	0
    unsavedxp 	0
    shadowbreedtemplate 	0
    shadowbreed 	0
    lastperkresettime 	0
    socialstatus 	0
    playeroptions 	0
    tempsaveteamid 	0
    tempsaveplayfield 	0
    tempsavex 	0
    tempsavey 	0
    PVPDuelKills 	0
    PVPDuelDeaths 	0
    PVPProfessionDuelKills 	0
    PVPProfessionDuelDeaths 	0
    PVPRankedSoloKills 	0
    PVPRankedSoloDeaths 	0
    PVPRankedTeamKills 	0
    PVPRankedTeamDeaths 	0
    PVPSoloScore 	0
    PVPTeamScore 	0
    PVPDuelScore 	0
    unreadmailcount 	0
    lastmailchecktime 	1283065897
    savedxp 	0
    features 	6
    apartmentsallowed 	1
    apartmentsowned 	0
    lastconcreteplayfieldinstance 	0
    mapoptions 	0
    mapareapart1 	0
    mapareapart2 	0
    mapareapart3 	0
    mapareapart4 	0
    missionbits1 	0
    missionbits2 	0
    missionbits3 	0
    missionbits4 	0
    missionbits5 	0
    missionbits6 	0
    missionbits7 	0
    missionbits8 	0
    missionbits9 	0
    missionbits10 	0
    missionbits11 	0
    missionbits12 	0
    sessiontime 	1234567890
    autoattackflags 	5
    personalresearchlevel 	0
    globalresearchlevel 	0
    personalresearchgoal 	0
    globalresearchgoal 	0
    battlestationside 	0
    battlestationrep 	10
    members 	999
    veteranpoints 	0
    monthspaid 	0
    PaidPoints 	0
    xpkillrange 	5
    inplay 	0
    attitude 	0
    alignment 	0
    icon 	0
    mesh 	17530
    deadtimer 	0
    lastsavexp 	0
    currentmass 	0
    itemtype 	0
    previoushealth 	50
    currentstate 	0
    age 	0
    attackspeed 	5
    metatype 	0
    rp 	0
    specialcondition 	1
    sk 	0
    clanredeemed 	0
    clanconserver 	0
    clandevoted 	0
    otunredeemed 	0
    otoperator 	0
    otfollowers 	0
    gos 	0
    clanvanguards 	0
    ottrans 	0
    clangaia 	0
    otmed 	1234567890
    clansentinels 	0
    otarmedforces 	0
    playerid 	1234567890
    alienlevel 	0
    aliennextxp 	1500
    alienxp 	0
    insurancepercentage 	0
    prevmovementmode 	3
    teamside 	0
    beltslots 	0
    absorbprojectileac 	0
    absorbmeleeac 	0
    absorbenergyac 	0
    absorbchemicalac 	0
    absorbradiationac 	0
    absorbcoldac 	0
    absorbnanoac 	0
    absorbfireac 	0
    absorbpoisonac 	0
    temporaryskillreduction 	0
    insurancetime 	0
    mapflags 	0
    changesidecount 	0
    procinitiative2 	3
    procchance2 	12
    procnano2 	222985
    damageoverridetype 	0
    reflectreturnedenergyac 	0
------------------------------------

.getfull succeeded - Listed all stats for Tinkerkeep.

```

**Comment by RaggyLP** [comment]
Unable to repro with the above steps, let it run for a bit to test, slowly killing myself with the proc lol.

![image](https://github.com/project-rubika/issues/assets/156821187/398f16c5-2792-4c4c-9f7a-b919dbb3b9fd)

This might've been fixed during the changes on #597. Please re-test :)

**Comment by TheTinkeringIdiot** [comment]
Unfortunately no, this issue continues. 

I noticed in further testing that as the auto-attack cycle locks, I get hit by 799 points of "unknown damage". This is exactly half the normal 1598 hit that Evangelical Reaper causes. This makes sense as I'm fighting a guard and thus flagged for PVP - a player (me) is hitting a player (also me) and so damage is halved. 

![image](https://github.com/project-rubika/issues/assets/96996793/826bbc6e-0285-4b34-b8f6-50e41d11c55d)

I repeated this test non-flagged against a normal non-PVP mob, and got the same half-damage hit after the auto-attack lockup.
